src/Raven.Client/Util/StreamWithTimeout.cs

```diff
@@ -154,9 +165,25 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
 
         private async Task<int> ReadAsyncWithTimeout(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            _readCts?.Dispose();
-            _readCts = cancellationToken == default ? new CancellationTokenSource() : CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-            _readCts.CancelAfter(_readTimeout);
+            if (_readCts == null)
+            {
+                _readCts = cancellationToken == default ? new CancellationTokenSource() : CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
you are not calling CancelAfter in this flow

Reply:
done